src/pal/pal_posix.h

```diff
@@ -48,14 +50,17 @@ namespace snmalloc
       puts("");
       fflush(stdout);
     }
+#endif
 
     /**
      * Report a fatal error an exit.
      */
     static void error(const char* const str) noexcept
     {
       puts(str);
+#ifdef SNMALLOC_HAS_EXECINFO_H
```

Comment:
Please remove these guards and move the guards in the callee inside the function.  The compiler will then elide this call.